### ISSUE TITLE
Failed connect to github.com:443 when behind Proxy

### ISSUE THREAD

**Issue by ericrange** [Glavin001/atom-beautify#117]
npm ERR! git clone https://github.com/Glavin001/Coffee-Formatter.git fatal: unable to access 'https://github.com/Glavin001/Coffee-Formatter.git/': Failed connect to github.com:443; No error

all the time i get this error. i CAN install other packages, but not atom-beautify cause of this shit formatter package

**Comment by jkanekunga** [comment]
"Connect to api.github.com:443 [api.github.com/192.30.252.126] failed: connect timed out"
Every time I want to share my work via Phycham in GitHub I got the above error. I used to do that before but after deleting my repository and trying to recreate via 'Share project on github', it keeps on giving that error.

**Comment by ericrange** [comment]
wut?? whats the connection with this repo?

**Comment by SoledadVac** [comment]
not fix yet

**Comment by elangoyuva** [comment]
Facing Same Issue.

**Comment by mvnnn** [comment]
same issue.

**Comment by wuming79** [comment]
same issue. I'm behind a proxy.

> git clone https://github.com/mininet/mininet.git
> Cloning into 'mininet'...
> fatal: unable to access 'https://github.com/mininet/mininet.git/': Failed to connect to github.com port 443: Connection timed out

**Comment by szeck87** [comment]
@wuming79 Try setting strict-ssl to false:  https://github.com/atom/apm#behind-a-firewall

**Comment by Preeti2309** [comment]
(C:\Users\Administrator\Anaconda3) C:\Users\Administrator\Desktop\Mask_RCNN-mast
er>git clone https://github.com/matterport/Mask_RCNN.git
Cloning into 'Mask_RCNN'...
fatal: unable to access 'https://github.com/matterport/Mask_RCNN.git/': Failed t
o connect to github.com port 443: Connection refused

**Comment by e-mubaraq** [comment]
You need to configure Git to use your proxy settings by running this command in your terminal or command line.

`git config --global http.proxy http://proxyuser:proxypwd@proxy.server.com:8080`

Where proxyuser and proxypwd are your credentials, and proxy.server.com:8080 is the server name and port you connect to.

**Comment by Chameena** [comment]
> 
> 
> You need to configure Git to use your proxy settings by running this command in your terminal or command line.
> 
> `git config --global http.proxy http://proxyuser:proxypwd@proxy.server.com:8080`
> 
> Where proxyuser and proxypwd are your credentials, and proxy.server.com:8080 is the server name and port you connect to.

This solved my issue

**Comment by mohammadyacn314** [comment]
just disable antivirus & firewall or 
convert https to http

**Comment by ArcLightSlavik** [comment]
`git config --global http.proxy http://proxyuser:proxypwd@proxy.server.com:8080`

Didn't work for me on ubuntu 18.02

`git config --global --unset http.proxy`

Worked well

**Comment by IYERADavid** [comment]
me also
git config --global --unset http.proxy
worked well

**Comment by bakasajoshua** [comment]
Had a similar issue, did an nslookup on github.com and that sorted me out.
`nslookup github.com`

**Comment by mahesh163051** [comment]
fatal: unable to access 'https://github.com/myrepository/': Failed to connect to github.com port 443: Timed out


Could some one please help me out

Thanks in advance

**Comment by Nambers** [comment]
same issue, but look like we can use SSH link instead of https to clone

**Comment by nelsonfinney** [comment]
> You need to configure Git to use your proxy settings by running this command in your terminal or command line.
> 
> `git config --global http.proxy http://proxyuser:proxypwd@proxy.server.com:8080`
> 
> Where proxyuser and proxypwd are your credentials, and proxy.server.com:8080 is the server name and port you connect to.

This worked for me

**Comment by mikedalu** [comment]
This worked for me also
 git config --global --unset http.proxy

**Comment by amitleuva1987** [comment]
This worked for me also
git config --global --unset http.proxy

**Comment by GursimranjeetSingh** [comment]
@amitleuva1987 where to use this command? command prompt ?
btw i am on windows 10

**Comment by amitleuva1987** [comment]
yes, hit this command on command prompt.

Btw, when I faced the issue, it was a temporary block from ISP. My ISP blocked certain websites, that was the reason that I was not able to use git commands. After few hours, it was resolved. 

I don't really need to use that command.  I would suggest you to check the same. If its' an ISP issue, you would also face issues surfing github.com over the browser.
